src/core/reference/include/openvino/reference/fake_quantize.hpp

```diff
@@ -55,133 +110,451 @@ void fake_quantize(const T* const arg,
+     * Overview:
+     *  Numpy broadcasted input tensors can be partitioned into two: outer and inner part (which also defines inner
+     *  stride as a product of inner part), so N-dimensional tensors can be processed using two loops.
+     *
+     *  For example with two inputs [2, 2, 3, 4] and [1, 1, 3, 4] we can have:
+     *      input 1 with shape [2, 2, 3, 4] can be divided into outer part [2, 2] and inner part [4, 5]
+     *      with inner stride = 12 (3 * 4) input 2 with shape [1, 1, 3, 4] can be divided into outer part [1, 1]
+     *      and inner part [4, 5] with inner stride = 12 (3 * 4)
+     *
+     *      Having that, those inputs can be processed by the following:
+     *
+     *      output_shape = {2, 2, 3, 4};
+     *      output_inner_stride = 12;
+     *      for (i = 0; i < shape_size(shape); i += output_inner_stride) {
+     *          first_input_stride = i;
+     *          second_input_stride = 0;
+     *          for (j = 0; j < 12; j++) {
+     *              *out++ = f(first_input[first_input_stride + j], second_input[second_input_stride + j]);
+     *          }
+     *      }
+     *
+     * ---------------------------------------------------
+     * How the partitioning is done:
+     *  Partitioning process starts with the last dimension of input tensor shape and it stops when either one of below
+     *  occurs:
+     *      - if the last dimension is equal to 1, partitioning stops at the dimension that is greater than 1 (this
+     *        dimension is not included in the inner part),
+     *      - if the last dimension is greater than 1, partitioning stops at the dimension that is equal to 1 (this
+     *        dimension is not included in the inner part).
+     *
+     *  Examples:
+     *      tensor_shape=[2, 3, 4, 5], inner_part = [2, 3, 4, 5], inner_stride = 120
+     *      tensor_shape=[1, 1, 4, 5], inner_part = [4, 5], inner_stride = 20
+     *      tensor_shape=[2, 3, 1, 1], inner_part = [1, 1], inner_stride = 1
+     *
+     *
+     * ---------------------------------------------------
+     * How the output inner stride is calculated:
+     *  Inner part (and inner stride) for every input tensor is determined. Then the size of output inner part is the
+     *  size of inner part with the fewest number of dimensions.
+     *
+     *  Example with 5 inputs:
+     *      input 1 shape [2, 3, 4, 5], inner_part = [2, 3, 4, 5], inner_stride = 120
+     *      input 2 shape [1, 3, 4, 5], inner_part = [3, 4, 5], inner_stride = 60
+     *      input 3 shape [2, 3, 1, 1], inner_part = [1, 1], inner_stride = 1
+     *      input 4 shape [2, 1, 1, 1], inner_part = [1, 1, 1], inner_stride = 1
+     *      input 5 shape [1, 1, 1, 1], inner_part = [1, 1, 1, 1], inner_stride = 1
+     *
+     *      output shape [2, 3, 4, 5], inner_part = [4, 5], inner_stride = 20
+     *
+     *      Inner part with fewest number of elements is [1, 1] for input 2. So the inner part for output shape is [4, 5]
```

Comment:
> Inner part with fewest number of elements is [1, 1] for input 2

Input 2 has other inner part:
`input 2 shape [1, 3, 4, 5], inner_part = [3, 4, 5]`.
Did you mean input 3 maybe?
`input 3 shape [2, 3, 1, 1], inner_part = [1, 1]`
Or we can numerate inputs starting with `0`.

Reply:
> Did you mean input 3 maybe?

yes, changed that to 3